types/config.d.ts

```diff
@@ -0,0 +1,38 @@
+interface dataFormat{
+    keyType : string,
+    valueType : string
+}
+//TODO : Add option for key type and mediatype among the three
+
+interface authentication{
+    enabled: boolean,
+    saslMechanism: string,
+    userName: string,
+    password: string,
+    serverName: string,
+}
+
+export interface entry{
+    key: mediaType,
+    value: mediaType
+}
+
+export interface statsItem{
+    STAT_NAME : string,
+    STAT_VALUE : string
+}
+
+export type mediaType = string | number | object ;
```

Comment:
Yeah, it also can be done. But I think it's better to provide the user with the type for mediatype. Although in client level the entry types will be of `any` type. This will make it easier for the user to make his own types of entries.

Reply:
It's not clear to me how `mediaType = any` is intended to be used. How it can simplify things for the user?

Reply:
At the client level, there will be no restriction on the type of media. But IMO if we export the mediaType with proper type definition then the user will have a reference of the media type, so that he can extend it to make its own,

Reply:
mmm, yeah it could make sense. I guess we need two different mediatypes for key and value

Reply:
It's not clear to me, why there is a need for different mediatypes for key and value.

Reply:
Yeah, TBH on second thought, it's not clear to me how to use the `mediaType` object type.
Let's try with a use case: 
Ideally key and value could be of different types, suppose the user wants to work with a cache with entries `(key: number, value: protobuf)`.
Starting from the current client definition:
`get(key: config.mediaType): Promise<config.mediaType>`
since key and value refer to the same object type, how the user could enforce different type checking?

Reply:
Yeah, in the above case if the user provides {key: number, value: JSON } as entries, then the ts compiler will not throw any error but the code will throw an error.
I think more discussion is needed on this. 
What if, we only internally type check on the key and value, and the user will have its own type defn for the entries?

Reply:
What kind of type checking could we enforce? Since key and value could be of any type, I'm not able to imagine a static type checking of some value.

Reply:
Like if the data format is `application/x-protostream` then the type of key/value should be of the `root` object.
Also if it's possible to dynamically change the type of key/value according to the data format entered by the user, then it will be a great solution.

Reply:
I'm afraid is something hard to do since Typescript provides only static type checking, so we cannot make any assumption on the runtime/dynamics values.

I would propose something different based on `typescript generics': I'm thinking of define key and value types as generic type for the client interface so to enforce the check that all the function methods are used accordingly.
Something like:
```
interface client<keyType, valueType> {
    connect(): Promise<any>,
    disconnect(): Promise<any>,
    get(key: keyType): Promise<valueType>,
    put(key: keyType, value: valueType, options?: object): Promise<valueType>,
```
wdyt?

More info on generics:
https://www.typescriptlang.org/docs/handbook/2/generics.html
https://www.typescriptlang.org/docs/handbook/2/generics.html#generic-types

Reply:
It's a great idea to implement!
I will try to change the PR according to this.